Source/Particles/Pusher/UpdatePosition.H

```diff
@@ -0,0 +1,28 @@
+#ifndef WARPX_PARTICLES_PUSHER_UPDATEPOSITION_H_
+#define WARPX_PARTICLES_PUSHER_UPDATEPOSITION_H_
+
+#include <AMReX_REAL.H>
+
+/* \brief Push the particle's positions over one timestep,
+ *    given the value of its momenta `ux`, `uy`, `uz` */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void UpdatePosition(
+    amrex::Real& x, amrex::Real& y, amrex::Real& z,
+    const amrex::Real ux, const amrex::Real uy, const amrex::Real uz,
+    const amrex::Real dt )
+{
+
+    constexpr amrex::Real inv_c2 = 1./(PhysConst::c*PhysConst::c);
+
+    // Compute inverse Lorentz factor
+    const amrex::Real inv_gamma = 1./std::sqrt(1. + (ux*ux + uy*uy + uz*uz)*inv_c2);
+    // Update positions over one time step
+    x += ux * inv_gamma * dt;
```

Comment:
As discussed offline, that would not work to push photons, for instance. But I am not sure it is worth looking at something general enough to encapsulate all particle types.

Reply:
OK, good point.
One way to have things work for photons is to define `ux`, `uy`, `uz` as the full momentum (i.e. in kg.m/s) **inside the code** (not just in the output). In this case we could do:
```
    const amrex::Real inv_gamma_m = 1./std::sqrt( m + (ux*ux + uy*uy + uz*uz)*inv_c2);
    // Update positions over one time step
    x += ux * inv_gamma_m * dt;
```
And this would work for photons (m=0).
But, as discussed, this is orthogonal to this PR.